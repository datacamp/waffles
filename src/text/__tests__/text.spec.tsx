import { render } from '@testing-library/react';

import { Text } from '../index';

describe('Text', () => {
  it('renders a span element containing the text', () => {
    const { container } = render(<Text>Taylor Swift Fanclub</Text>);

    const text = container.querySelector('span');

    expect(text).toBeInTheDocument();
    expect(text).toHaveTextContent('Taylor Swift Fanclub');
  });

  it('sets the data attribute on the text element', () => {
    const { getByTestId } = render(
      <Text data-testid="taylors-text">Taylor Swift Fanclub</Text>,
    );

    const text = getByTestId('taylors-text');

    expect(text).toBeInTheDocument();
  });

  it('renders an element passed via "as" prop', () => {
    const { container } = render(<Text as="p">Taylor Swift Fanclub</Text>);

    const text = container.querySelector('p');

    expect(text).toBeInTheDocument();
    expect(text).toHaveTextContent('Taylor Swift Fanclub');
  });

  it('renders snapshot', () => {
    const { getByText } = render(<Text>Definitely not Taylor Swift</Text>);

    const text = getByText(/taylor swift/i);

    expect(text).toBeInTheDocument();
    expect(text).toMatchSnapshot();
  });
});
