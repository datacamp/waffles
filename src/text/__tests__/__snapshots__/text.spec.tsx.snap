// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text renders snapshot 1`] = `
.emotion-0 {
  color: #05192D;
  font-family: Studio-Feixen-Sans,Arial,sans-serif;
  font-size: 14px;
  font-weight: 400;
  line-height: 1;
  margin: 0;
  padding: 0;
}

<span
  class="emotion-0"
>
  Definitely not Taylor Swift
</span>
`;
